rust/tool/src/utils.rs

```diff
@@ -47,3 +51,16 @@ impl SecureTempDirExt for TempDir {
         Ok(path)
     }
 }
+
+/// Generate a random name for a temporary file.
+///
+/// This is heavily inspired by the way temporary names are generated in the `tempfile` crate.
+/// Since the `tempfile` crate does not expose this functionality, we have to recreate it here.
+pub fn tmpname() -> OsString {
+    let mut buf = OsString::with_capacity(TEMPFILE_RANDOM_LENGTH);
+    let mut char_buf = [0u8; 4];
+    for c in repeat_with(fastrand::alphanumeric).take(TEMPFILE_RANDOM_LENGTH) {
```

Comment:
This is used in `write_secure_file` and the second line in the `fastrand` [docs](https://docs.rs/fastrand/1.9.0/fastrand/) says: "The implementation uses [Wyrand](https://github.com/wangyi-fudan/wyhash), a simple and fast generator but not cryptographically secure."

In our usecase this is not important, but I would at least mention this in a comment or the `tmpfile` docstring.

**That being said**, aren't you reimplementing [`NamedTempFile::new_in`](https://docs.rs/tempfile/3.3.0/tempfile/struct.NamedTempFile.html#method.new_in) and then calling `keep()` on it? This would avoid adding yet another dependency.

Reply:
I wish it was that easy, but `NamedTempFile` already creates a file. So to implement our "secure" tempfiles, we would need to chmod this tempfile after having created it. Additionally, sometimes we actually only need a path without creating the file ourselves (e.g. when calling objcopy). That's why `tmpname()` is `pub`

Reply:
Mmmh. Too bad.